tools/points-service/main.go

```diff
@@ -0,0 +1,529 @@
+					},
+				),
+
+				// VaultDeregistered => bulk opt-out
+				events.NewEventHandler(
+					"VaultDeregistered",
+					func(ev *middleware.MevcommitmiddlewareVaultDeregistered, blockNumber uint64) {
+						vaultAddr := ev.Vault.Hex()
+						// If no 'vault' column, you'd do a different approach.
+						rows, err := db.Query(`
+                            SELECT pubkey, adder
+                            FROM events
+                            WHERE vault = ?
+                              AND opted_out_block IS NULL
+                        `, vaultAddr)
+						if err != nil {
+							logger.Error("failed to query for vault", "error", err)
+							return
+						}
+						defer rows.Close()
+
+						for rows.Next() {
+							var pubkey, adder string
+							if err := rows.Scan(&pubkey, &adder); err != nil {
+								logger.Error("scan error", "error", err)
+								continue
+							}
+							insertOptOut(db, logger, pubkey, adder, "VaultDeregistered", blockNumber)
+						}
+					},
+				),
+				// OperatorDeregistered => bulk opt-out
+				events.NewEventHandler(
+					"OperatorDeregistered",
+					func(ev *middleware.MevcommitmiddlewareOperatorDeregistered, blockNumber uint64) {
+						operatorAddr := ev.Operator.Hex()
+						rows, err := db.Query(`
+                            SELECT pubkey
+                            FROM events
+                            WHERE adder = ?
+                              AND opted_out_block IS NULL
+                        `, operatorAddr)
+						if err != nil {
+							logger.Error("failed to query for operator", "error", err)
+							return
+						}
+						defer rows.Close()
+
+						for rows.Next() {
+							var pubkey string
+							if err := rows.Scan(&pubkey); err != nil {
+								logger.Error("scan error", "error", err)
+								continue
+							}
+							insertOptOut(db, logger, pubkey, operatorAddr, "OperatorDeregistered", blockNumber)
+						}
+					},
+				),
+				// ValRecordDeleted => single validator removal
+				events.NewEventHandler(
+					"ValRecordDeleted",
+					func(ev *middleware.MevcommitmiddlewareValRecordDeleted, blockNumber uint64) {
+						pubkeyHex := common.Bytes2Hex(ev.BlsPubkey)
+
+						// find adder
+						var adderHex string
+						err := db.QueryRow(`
+                            SELECT adder FROM events
+                            WHERE pubkey = ?
+                              AND opted_out_block IS NULL
+                            LIMIT 1
+                        `, pubkeyHex).Scan(&adderHex)
+						if err != nil {
+							logger.Error("failed to find active adder", "error", err, "pubkey", pubkeyHex)
+							return
+						}
+						insertOptOut(db, logger, pubkeyHex, adderHex, "ValRecordDeleted", blockNumber)
+					},
+				),
+			}
+
+			// 6. Subscribe
+			sub, err := listener.Subscribe(handlers...)
+			if err != nil {
+				return fmt.Errorf("subscribe error: %w", err)
+			}
+			defer sub.Unsubscribe()
+
+			// 7. Start daily routine to recompute points
```

Comment:
This service does not handle clean shutdown. Check other services to see how to do that. There should be some context and also signals (SIGINT/SIGTERM) should be handled. All the other services should have this functionality.